webview-ui/src/utils/extract-command-pattern.ts

```diff
@@ -0,0 +1,315 @@
+/**
+ * Extracts a generalizable command pattern from a specific command.
+ * This function creates patterns that can be used for whitelisting similar commands.
+ *
+ * Examples:
+ * - "npm test" -> "npm test"
+ * - "npm run build" -> "npm run"
+ * - "git commit -m 'message'" -> "git commit"
+ * - "echo 'hello world'" -> "echo"
+ * - "python script.py --arg value" -> "python"
+ * - "./scripts/deploy.sh production" -> "./scripts/deploy.sh"
+ * - "cd /path/to/dir && npm install" -> "cd * && npm install"
+ * - "rm -rf node_modules" -> "rm"
```

Comment:
I think in this specific example I'd prefer the exact rm command; I would never fully whitelist rm